include/zephyr/spinlock.h

```diff
@@ -105,6 +107,28 @@ bool z_spin_lock_mem_coherent(struct k_spinlock *l);
  */
 typedef struct z_spinlock_key k_spinlock_key_t;
 
+static ALWAYS_INLINE void z_spinlock_validate_pre(struct k_spinlock *l)
+{
+	ARG_UNUSED(l);
+#ifdef CONFIG_SPIN_VALIDATE
+	__ASSERT(z_spin_lock_valid(l), "Invalid spinlock %p", l);
+#ifdef CONFIG_KERNEL_COHERENCE
+	__ASSERT_NO_MSG(z_spin_lock_mem_coherent(l));
+#endif
+#endif
```

Comment:
Not really the fault of this patch, but this clearly some #if'ery refactoring.  Testing the kconfig inside a function named "validate" seems silly.  Likewise we should just define z_spin_lock_mem_coherent() to a nop if !COHERENCE.

Reply:
No, it was just a drive-by comment.  But feel free if you want to. :)

Reply:
I would, but meetings (on no-meeting day no less!)